internal/repositories/urls/memory_repository.go

```diff
@@ -0,0 +1,62 @@
+package urls
+
+import (
+	"sync"
+
+	"github.com/izaake/go-shortener-tpl/internal/models"
+)
+
+type memoryRepository struct {
+	Users   map[string]map[string]string
+	BaseURL string
+}
+
+var (
+	lock = sync.RWMutex{}
+)
+
+func NewMemoryRepository(baseURL string) Repository {
+	Users := make(map[string]map[string]string)
+	return &memoryRepository{Users: Users, BaseURL: baseURL}
+}
+
+func (r memoryRepository) Save(user *models.User, ignoreConflicts bool) error {
+	if r.Users[user.ID] == nil {
+		r.Users[user.ID] = map[string]string{}
+	}
+
+	for _, url := range user.URLs {
+		lock.Lock()
+		r.Users[user.ID][url.ShortURL] = url.OriginalURL
+		lock.Unlock()
+	}
```

Comment:
Блокировка - довольно тяжеловесная функция, ее следует выносить за границы цикла.
```suggestion
	lock.Lock()
	for _, url := range user.URLs {
		r.Users[user.ID][url.ShortURL] = url.OriginalURL
	}
	lock.Unlock()
```